src/store/groups.js

```diff
@@ -50,6 +57,12 @@ const mutations = {
 				})
 			}
 		})
+		state.sortedGroups = state.groups
+			.map(v => {
```

Comment:
what is this?
I don't understand what you're trying to do with toString.
Also please use proper variables names instead of letters like `v` :wink:

Reply:
Also, it could be nicer to have a dedicated `sortGroups`  mutation which you can fire directly
That would avoid duplicate code :grin:

Reply:
can mutations call another mutation?

Reply:
there is already a sortGroups mutation but you cannot call it from another mutation from what im reading, so we can just add a new commit("sortGroups") to every call that removes contacts from groups, remove groups and add contacts to groups but is that good?

Reply:
> can mutations call another mutation?

yes, but this is not as clean as calling it from the action after the first mutation I would say (but be careful about async code)

Reply:
What i do with toString is use the default .sort() (the default sort function is used when there are no arguments passed), that means it sorts the array based on the utf-16 code, which means that i dont need to do any sorting function, just give the .sort() default function a way to get the name of each group
You can see that MDN says that in [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort)

Reply:
why not return the name directly?

```suggestion
			.map(group => group.name) 
```

Reply:
Ah right, because then you'll sort the array of names.
Make sense.

I still find the syntax hard to understand :thinking:

Reply:
got any ideas for making the map more understandable? i know there is 
```js
{...group, toString: () => group.name}
// or
Object.assign(group, { toString: group.name })
```

Reply:
Ok, check latest commit for the sortGroups part, the map is still for discussion, i showed 2 options that are the alternatives of the current one